charts/argoconfig/templates/application.yaml

```diff
@@ -1,19 +1,25 @@
-{{- if .Values.application.enabled }}
+{{- $defaults := .Values.apps.defaults -}}
+{{- range $name, $app := .Values.apps -}}
+{{- if $app.enabled -}}
 apiVersion: argoproj.io/v1alpha1
 kind: Application
 metadata:
-  name: {{ include "argoconfig.fullname" . }}
-  namespace: {{ .Values.application.namespace }}
+  name: {{ include "argoconfig.fullname" $ }}-{{ $name }}
+  {{- with (default $defaults.namespace $app.namespace) }}
```

Comment:
noob question here, is this gonna set `$defaults.namespace` if no `$app.namespace` is defined? If yes, it's very elegant imho.

Reply:
That's what `default` is doing, yeah. It's also using `with` to ensure that no namespace is set if neither `$defaults.namespace` and `$app.namespace` is set.